drafts/str4d-stratum/draft1.rst

```diff
@@ -0,0 +1,256 @@
+
+- ``CONNECT_HOST`` (str): The host string used by the miner.
+
+  - ``pool.example.com``
+
+- ``CONNECT_PORT`` (int): The port used by the miner.
+
+  - ``3337``
+
+- ``MINER_USER_AGENT`` (str): A free-form string specifying the type and version
+  of the mining software. Recommended syntax is the User Agent format used by
+  Zcash nodes.
+
+  - ``Zatoshi/1.0.0``
+
+- ``SESSION_ID`` (str): The id for a previous session that the miner wants to
+  resume (e.g. after a temporary network disconnection).
+
+  - MAY be ``null`` indicating that the miner wants to start a new session.
+
+Response::
+
+    {"id": 1, "result": ["NONCE_1", "SESSION_ID"], "error": null}\n
+
+- ``SESSION_ID`` (str): The session id, for use when resuming.
+
+  - MAY be ``null`` indicating that the server does not support session
+    resuming.
+
+- ``NONCE_1`` (hex): The first part of the block header nonce.
+
+The nonce in Zcash's block header is 32 bytes long. The miner MUST pick
+``NONCE_2`` such that ``len(NONCE_2) = 32 - len(NONCE_1)`` in bytes, or
+``len(NONCE_2) = 64 - len(NONCE_1)`` in hex.
+
+
+``mining.authorize()``
+----------------------
+
+Request::
+
+    {"id": 2, "method": "mining.authorize", "params": ["WORKER_NAME", "WORKER_PASSWORD"]}\n
+
+- ``WORKER_NAME`` (str): The worker name.
+
+- ``WORKER_PASSWORD`` (str): The worker name.
+
+Response::
+
+    {"id": 2, "result": AUTHENTICATED, "error": null}\n
+
+- ``AUTHENTICATED`` (bool): The result of authentication.
+
+The server MAY provide an error message on invalid authentication.
+
+``mining.set_target()``
+-----------------------
+
+Server message::
+
+    {"id": null, "method": "mining.set_target", "params": ["TARGET"]}\n
+
+- ``TARGET`` (hex): The server target for the next received job and all
+  subsequent jobs (until the next time this message is sent).
+
+  - Miners SHOULD NOT submit work above this target.
+
+  - Servers SHOULD NOT accept submissions above this target.
+
+When displaying the current target in the UI to users, miners MAY convert the
+target to an integer difficulty as used in Bitcoin miners. When doing so, miners
+SHOULD use ``powLimit`` (as defined in ``src/chainparams.cpp``) as the basis for
+conversion.
+
+``mining.notify()``
+-------------------
+
+Server message::
+
+    {"id": null, "method": "mining.notify", "params": ["JOB_ID", "VERSION", "PREVHASH", "MERKLEROOT", "RESERVED", "TIME", "BITS", CLEAN_JOBS]}\n
+
+- ``JOB_ID`` (str): The id of this job.
+
+- ``VERSION`` (int): The block header version. Used as a switch for subsequent
+  parameters.
+
+The following parameters are only valid for ``VERSION == 4``:
+
+- ``PREVHASH`` (hex): The hash of the previous block.
+
+- ``MERKLEROOT`` (hex): The Merkle root of the transactions in this block.
+
+- ``RESERVED`` (hex): A 256-bit reserved field; zero by convention.
+
+- ``TIME`` (hex): The block time suggested by the server.
```

Comment:
I'm not entirely clear on whether the convention with original Stratum was to use hex strings or integers for `TIME`.